lib/ansible/cli/adhoc.py

```diff
@@ -153,7 +153,7 @@ def run(self):
         elif self.options.one_line:
             cb = 'oneline'
         else:
-            cb = 'minimal'
+            cb = 'minimal' if C.DEFAULT_STDOUT_CALLBACK == 'default' else C.DEFAULT_STDOUT_CALLBACK
```

Comment:
So custom `stdout_callback` is picked up only with enabled `bin_ansible_callbacks`.

Done in https://github.com/ansible/ansible/pull/26098/commits/d8afdcd47b69a3c207d8446f351b41103f3eed50